PROVESFlightControllerReference/Components/TcSecurityDeframer/TcSecurityDeframer.cpp

```
@@ -150,35 +191,219 @@ void TcSecurityDeframer ::SET_SEQ_NUM_cmdHandler(FwOpcodeType opCode, U32 cmdSeq
+    // against the store that is actually on flash.
+    (void)this->loadKeyStore();
+
+    // PROVISION_KEY is trust-on-first-use bootstrap: only honored while the store is empty.
+    // Once any key exists, rotation must go through ADD_KEY/REMOVE_KEY (which require auth).
+    if (this->activeKeyCount() != 0) {
+        this->log_WARNING_HI_KeyProvisionFailed(KeyStoreProvisionStatus::NotEmpty);
+        this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::EXECUTION_ERROR);
+        return;
+    }
+
+    uint8_t keyBytes[Ccsds355_0_B_2::kTCSecurityTrailer];
```

Clang (14.0.6)

[warning] do not declare C-style arrays, use std::array<> instead

(modernize-avoid-c-arrays)